backend/app/src/test/kotlin/io/tolgee/api/v2/controllers/UserMfaControllerTest.kt

```diff
@@ -117,28 +118,30 @@ class UserMfaControllerTest : AuthorizedControllerTest() {
 
   @Test
   fun `it invalidates tokens generated prior a mfa status change`() {
-    loginAsAdminIfNotLogged()
-    Thread.sleep(2000)
-
-    val enableRequestDto = UserTotpEnableRequestDto(
-      totpKey = TOTP_KEY,
-      otp = mfaService.generateStringCode(encodedKey),
-      password = initialPassword
-    )
-
-    performAuthPut("/v2/user/mfa/totp", enableRequestDto).andIsOk
-    refreshUser()
-    performAuthGet("/v2/user").andExpect(MockMvcResultMatchers.status().isUnauthorized)
-
-    logout()
-    loginAsAdminIfNotLogged()
-    Thread.sleep(2000)
-
-    val disableRequestDto = UserTotpDisableRequestDto(
-      password = initialPassword
-    )
-    performAuthDelete("/v2/user/mfa/totp", disableRequestDto).andIsOk
-    refreshUser()
-    performAuthGet("/v2/user").andExpect(MockMvcResultMatchers.status().isUnauthorized)
+    retry {
```

Comment:
All the tests should be wrapped in a retry block, due to the dependency to timing stuff. Ideally, we should look into cleaning up these timing-sensitive tests and inject a `Clock` object, so we can easily mock time and simulate huge time shifts (or "disable" time).

An internal endpoint to set the time would also be handy for the Cypress tests, so they can also manipulate the time, and avoid flaky tests.

This would be cleaner and also applies to token expiry tests. 🤔

Reply:
It's already possible to mock time in integrations tests. There is `io.tolgee.component.CurrentDateProvider` which can be used to mock the time. Using `@MockBean` I just didn't want to dig too deep here. Feel free, to fix it right, but for now I would just merge it as it is. 😄

Reply:
I'll open an issue about it 👍🏻 

I'm unsure the mfa service would respond to mocking that, it'll need to be checked